tests/NSubstitute.Documentation.Tests.Generator/DocumentationTestsGenerator.cs

```diff
@@ -37,7 +37,13 @@ public static string FormatCode(string code)
 
     private static string GenerateTestsClassName(AdditionalText markdownFile)
     {
-        return $"Tests_{Path.GetFileNameWithoutExtension(markdownFile.Path).Replace("-", "_")}";
+        var file = Path.GetFileNameWithoutExtension(markdownFile.Path);
+        var pathSegments = markdownFile.Path.Split(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
```

Comment:
you can avoid manually working with paths
```
var fileName = Path.GetFileNameWithoutExtension(markdownFile.Path);
var directoryName = new FileInfo(markdownFile.Path).Directory.Name;
var nameToUse = string.Equals(fileName, "index", StringComparison.InvariantCultureIgnoreCase) ? directoryName  : fileName;
return $"Tests_{nameToUse.Replace("-", "_")}";
```

Reply:
Applied suggestion :+1: